google/cloud/spanner/json.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_JSON_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_JSON_H
+
+#include "google/cloud/spanner/version.h"
+#include <ostream>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+/**
+ * A simple representation for the Spanner JSON type: a lightweight,
+ * text-based, language-independent data interchange format. JSON (the
+ * JavaScript Object Notation) defines a small set of formatting rules
+ * for the portable representation of structured data. See RFC 7159.
+ *
+ * A `JSON` value can be constructed from, and converted to a `std::string`.
+ * `JSON` values can be compared (by string) for equality, and streamed.
+ *
+ * There is no syntax checking of JSON strings in this interface. The user
+ * is expected to only construct `JSON` values from well-formatted strings.
+ */
+class JSON {
```

Comment:
I think this should be `Json`:


> prefer to capitalize abbreviations as single words, e.g., StartRpc() rather than StartRPC().

https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

Reply:
Done, not that I particularly like that rule, or at least when applied to a single word.